rules.py

```diff
@@ -826,7 +826,38 @@ def check(self):
         if error_log:
             raise base.ElectionTreeError(
                 "The Election File contains unreferenced Candidates", error_log)
-                
+
+class CheckIdentifiers(base.TreeRule):
+    """Check that the NIST objects in the feed has an '<ExternalIdentifier>' block.
+	Add error message if the block is missing."""
+    
+    identifier_values = {}
+    error_log = []
+    nist_objects = ('Candidate', 'Contest', 'Party')
+    def check(self):
+        for event, element in etree.iterwalk(self.election_tree):
+	    tag = self.strip_schema_ns(element)
+            if tag not in self.nist_objects:
+                continue
+            else:
+                nist_obj = tag
+                object_id = element.get('objectId')
+                value = element.find("ExternalIdentifiers").find("ExternalIdentifier").find("Value")
+                if not value.text:
+		    error_message = ("Missing value in <Value> tag for "
+			"{0} with ID {1}".format(nist_obj, object_id))
+                    self.error_log.append(base.ErrorLogEntry(value.sourceline, error_message))
```

Comment:
Because `error_log` is local to this function, you don't need to reference it with `self`.

Reply:
error_log is not local to the function. It is defined just above def. So referenced it using self.